File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/di/WordsViewComponent.java
package com.przemyslawlusnia.vocabularycreator.wordlist.di;

import com.przemyslawlusnia.vocabularycreator.core.di.WordsFragmentScope;
import com.przemyslawlusnia.vocabularycreator.wordlist.presentation.WordsFragment;
import dagger.Subcomponent;

@WordsFragmentScope
@Subcomponent(modules = WordsViewModule.class)
public interface WordsViewComponent {

  void inject(WordsFragment wordsFragment);

}

File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/di/WordsViewComponent.kt
package com.przemyslawlusnia.vocabularycreator.wordlist.di

import com.przemyslawlusnia.vocabularycreator.wordlist.presentation.WordsFragment
import dagger.Subcomponent

@WordsViewScope
@Subcomponent(modules = arrayOf(WordsViewModule::class))
interface WordsViewComponent {

    fun inject(wordsFragment: WordsFragment)

}